x-pack/plugins/security_solution/public/entity_analytics/components/include_closed_alerts_section.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { css } from '@emotion/react';
+import {
+  EuiSuperDatePicker,
+  EuiButton,
+  EuiText,
+  EuiFlexGroup,
+  EuiSwitch,
+  EuiFlexItem,
+  EuiBottomBar,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { useAppToasts } from '../../common/hooks/use_app_toasts';
+import * as i18n from '../translations';
+import { useConfigureSORiskEngineMutation } from '../api/hooks/use_configure_risk_engine_saved_object';
+
+export const IncludeClosedAlertsSection = ({
+  includeClosedAlerts,
+  setIncludeClosedAlerts,
+  from,
+  to,
+  onDateChange,
+}: {
+  includeClosedAlerts: boolean;
+  setIncludeClosedAlerts: (value: boolean) => void;
+  from: string;
+  to: string;
+  onDateChange: ({ start, end }: { start: string; end: string }) => void;
+}) => {
+  const [start, setFrom] = useState(from);
+  const [end, setTo] = useState(to);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showBar, setShowBar] = useState(false);
+  const { addSuccess } = useAppToasts();
+
+  // Retrieve state from localStorage on initial render
+  useEffect(() => {
+    const savedIncludeClosedAlerts = localStorage.getItem('includeClosedAlerts');
+    const savedStart = localStorage.getItem('dateStart');
```

Comment:
This component's name is "IncludeClosedAlertsSection," but it handles dates. That isn't very clear. Please rename the component or, even better, break it down into smaller components.